packages/data-schema/src/SchemaProcessor.ts

```diff
@@ -1325,12 +1497,13 @@ const schemaPreprocessor = (
   lambdaFunctions: LambdaFunctionDefinition;
   customSqlDataSourceStrategies?: CustomSqlDataSourceStrategy[];
 } => {
-  const gqlModels: string[] = [];
-
+  const gqlComponents: string[] = [];
   const customQueries = [];
   const customMutations = [];
   const customSubscriptions = [];
   let shouldAddConversationTypes = false;
+  const definedTypes = new Set<string>();
```

Comment:
I think our duplicates, which this set is meant to work out are caused by a logic error. I also think, as noted that we still have duplicates, but they have different names now.